_posts/2015-05-29-chrome-push-api.md

```diff
@@ -0,0 +1,98 @@
+
+### ServiceWorker Registration
+Now before we get started with registering you need to set up a Google Application that can be used with Google Cloud Messaging to actually send the notifications, it is free for development and takes about 2-3 minutes.
+
+#### Google Application Setup
+1. Go to [Google Developer Console](https://console.developers.google.com) and **create** a new application.
+2. Note your new applications **Project Number**:
+<img src="/img/blog/google-api-console-app-id.png" alt="Google API Project Number" class="img-responsive"/>
+3. Go to the new application and under **APIs & auth** click **APIs** and look for "messaging" and enable:
+  - *Google Cloud Messaging for Chrome*
+  - *Google Cloud Messaging for Android*
+4. Under the same sidebar menu, go to **Credentials** and generate new **Public API access** keys for **server**. Keep them handy for a little later on.
+
+#### manifest.json
+Google Chrome relies on a simple JSON file to detect your websites Google Cloud app. It is placed in the root of your website and it goes a little something like this:
+{% gist 81e37636d6b0a55788f1 manifest.json %}
+Replace `application_project_number` with the project number of your Google App from the steps above. Moving on!
+
+---
+
+#### ServiceWorker Life-Cycle
+ServiceWorkers have a registration process that goes through the following steps:
+
+- **install**: currently being installed
+- **waiting**: waiting on registration
+- **active**: registered and running
+
+---
+
+#### Scope
+ServiceWorkers also have a *scope* which is set to the directory under which it can be found. For example, if your service worker script is located at `/static/js/muh/scripts/ServiceWorker.js` Then its *scope* will be: `/static/js/muh/scripts/`. This will not work; your users will not be doing their browsing in your static assets, so placing this script further up the directory tree is highly recommended. Service Workers also will not install if not served over HTTPS.
+
+#### Registering
+Now that you know what to expect with setting up a suitable environment for your service worker let’s get it installed.
+
+Now, a good installation script should also consider that their might be a service worker already in place and running. A lot of other examples want to go straight to registering a service worker, but that will give you a bunch of redundant service workers. To check for that let’s do the following:
+{% gist 81e37636d6b0a55788f1 check-sw-reg.js %}
+Now that we have the installation status we know if we need to register a service worker or not. So let's say we do. Then we do this:
```

Comment:
* Rephrase of the sentence: "Now that we have the installation status: we know if registration of the service worker is necessary."
* Rephrase of sentence: "Assuming registration is necessary:"